packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2192,35 +2248,56 @@ class AnnotatedRegionLayer<T> extends ContainerLayer {
   /// If not provided, offset defaults to [Offset.zero].
   ///
   /// Ignored if [size] is not set.
+  ///
+  /// Unlike [OffsetLayer], this parameter [offset] does not affect the layer's
+  /// children.
   final Offset offset;
 
-  @override
-  S find<S>(Offset regionOffset) {
-    final S result = super.find<S>(regionOffset);
-    if (result != null)
-      return result;
-    if (size != null && !(offset & size).contains(regionOffset))
-      return null;
-    if (T == S) {
-      final Object untypedResult = value;
-      final S typedResult = untypedResult;
-      return typedResult;
-    }
-    return null;
-  }
+  /// Whether this layer should prevent layers visually behind it from being
+  /// hit during a hit test.
+  ///
+  /// If [opaque] is true, this layer will absorb the hit if this layer contains
+  /// the target position and has the requested annotation.
+  ///
+  /// If [opaque] is false, this layer will not change the absorption value
+  /// returned by its children.
+  ///
+  /// This defaults to false.
+  ///
+  /// See also:
+  ///
+  ///  * [HitTestBehavior], which controls similar logic when hit-testing in the
+  ///    render tree.
+  final bool opaque;
 
+  /// Adds annotations from the children, then from this layer.
+  ///
+  /// The children's annotations are always added.
+  ///
+  /// The layer's annotation [value] is added if the type `S` matches `T`, and
+  /// the target position meets the requirement defined by [size] and [offset].
+  ///
+  /// The return value is true if any child returns true, or the layer's
+  /// annotation is added and [opaque] is true.
```

Comment:
You should add a description of the `onlyFirst`, `result`, and `localPosition` parameters to these docs.  I like your description in the comment below, but the ASCII art won't work here.

Reply:
For `onlyFirst`, maybe something like:

The [onlyFirst] parameter indicates that the while walking the layer tree, the search will stop at the first child at [localPosition]. If [onlyFirst] is true, then the value of [opaque] is ignored. If [onlyFirst] is false, then [opaque] is used to decide whether or not the tree walk continues to the other children.  If [opaque] is false, then [findAnnotations] will return true if there is any layer at [localPosition].

The parent will return true if any of the children return true from their [findAnnotations].